packages/faustwp-cli/src/blockset.ts

```diff
@@ -0,0 +1,182 @@
+
+import { getWpUrl, getWpSecret, hasYarn } from './utils/index.js';
+import { infoLog } from './stdout/index.js';
+
+const ROOT_DIR = process.cwd();
+const FAUST_DIR = path.join(ROOT_DIR, '.faust');
+const FAUST_BUILD_DIR = path.join(FAUST_DIR, 'build');
+const BLOCKS_DIR = path.join(FAUST_DIR, 'blocks');
+const MANIFEST_PATH = path.join(BLOCKS_DIR, 'manifest.json');
+const IGNORE_NODE_MODULES = '**/node_modules/**';
+const FAUST_BLOCKS_SRC_DIR = 'wp-blocks';
+
+// Ensure required directories exist
+fs.ensureDirSync(BLOCKS_DIR);
+
+export type Manifest = {
+  blocks: any[];
+  timestamp: string;
+};
+
+const manifest: Manifest = {
+  blocks: [],
+  timestamp: new Date().toISOString(),
+};
+
+/**
+ * Fetches paths to all block.json files while ignoring node_modules.
+ *
+ * @returns {Promise<string[]>} An array of paths to block.json files.
+ */
+export async function fetchBlockFiles(): Promise<string[]> {
+  return glob(`${FAUST_BUILD_DIR}/**/block.json`, {
+    ignore: IGNORE_NODE_MODULES,
+  });
+}
+
+/**
+ * Processes each block.json file, copying its directory and updating the manifest.
+ *
+ * @param {string[]} files - An array of paths to block.json files.
+ * @returns {Promise<void>}
+ */
+export async function processBlockFiles(files: string[]): Promise<void> {
+  await fs.emptyDir(BLOCKS_DIR);
+  // Use Promise.all and map instead of for...of loop
+  await Promise.all(
+    files.map(async (filePath) => {
+      const blockDir = path.dirname(filePath);
+      const blockName = path.basename(blockDir);
+      const destDir = path.join(BLOCKS_DIR, blockName);
+
+      await fs.copy(blockDir, destDir);
+
+      const blockJson = await fs.readJson(filePath);
+      manifest.blocks.push(blockJson);
+    }),
+  );
+}
+
+/**
+ * Creates a ZIP archive of the blocks.
+ *
+ * @returns {Promise<string>} Path to the created ZIP archive.
+ */
+export async function createZipArchive(): Promise<string> {
+  const zipPath = path.join(FAUST_DIR, 'blocks.zip');
+  const output = fs.createWriteStream(zipPath);
+  const archive = archiver('zip');
+
+  archive.pipe(output);
+  archive.directory(BLOCKS_DIR, false);
+  await archive.finalize();
+
+  return zipPath;
+}
+
+/**
+ * Uploads the ZIP archive to WordPress.
+ *
+ * @param {string} zipPath - Path to the ZIP archive to be uploaded.
+ * @returns {Promise<void>}
+ */
+export async function uploadToWordPress(zipPath: string): Promise<void> {
+  if (!fs.existsSync(zipPath)) {
+    throw new Error('Provided zipPath does not exist.');
+  }
+
+  const form = new FormData();
+  form.append('zipfile', fs.createReadStream(zipPath));
+
+  const apiUrl = `${getWpUrl()}/wp-json/faustwp/v1/blockset`;
+  const headers = {
+    ...form.getHeaders(),
+    'x-faustwp-secret': getWpSecret() || '',
+  };
+
+  try {
+    const response = await fetch(apiUrl, {
+      headers,
+      method: 'POST',
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
Could we get away with not using a `@ts-ignore` here?

Reply:
@blakewilson I am 💯 in favor of removing it if we can find a way to resolve this TypeScript issue:
<img width="1153" alt="Screenshot 2023-10-12 at 4 38 31 PM" src="https://github.com/wpengine/faustjs/assets/6676674/c465ab14-7f6c-4207-b638-d206cd97f905">

Reply:
Will provide a fix.